.github/workflows/test.yml

```diff
@@ -2,76 +2,99 @@ name: test
-    steps:
-      - uses: actions/checkout@v4
-      - name: Build
-        run: CC=clang make all
-      - name: Run tests
-        run: ./build/test
-
-  windows:
-    name: windows
-    runs-on: windows-latest
-    steps:
-      - uses: actions/checkout@v4
-      
-      - name: Install MinGW
-        run: |
-          choco install mingw
-          echo "C:\tools\mingw64\bin" >> $GITHUB_PATH
-
-      - name: Build
-        run: CC=gcc make all
-        shell: bash
-        
-      - name: Run tests
-        run: ./build/test.exe
-        shell: bash
-
-  macos:
-    name: macos
-    runs-on: macos-latest
+  build:
+    name: Library - ${{ matrix.settings.name }}
+    runs-on: ${{ matrix.settings.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        settings:
+          - name: MacOS arm64
+            target: aarch64-apple-darwin
+            runner: macos-latest
+          - name: Linux x86
+            target: x86_64-unknown-linux-gnu
+            runner: ubuntu-latest
+          - name: Linux arm64
+            target: aarch64-unknown-linux-gnu
+            runner: ubuntu-latest
+          - name: Linux x86 using clang
+            target: x86_64-unknown-linux-gnu
+            runner: ubuntu-latest
+            cc: clang
+          - name: Linux arm64 using clang
+            target: aarch64-unknown-linux-gnu
+            runner: ubuntu-latest
+            cc: clang
+          - name: Windows x86
+            target: x86_64-pc-windows-msvc
+            runner: windows-latest
+          - name: Windows arm64
+            target: aarch64-pc-windows-msvc
+            runner: windows-latest
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Cross-Compile Build
+        if: ${{ matrix.settings.cc != '' }}
+        run: CC=${{ matrix.settings.cc }} make all
       - name: Build
+        if: ${{ !matrix.settings.cc }}
         run: make all
       - name: Run tests
         run: ./build/test
 
   rust-bindings:
-    runs-on: ubuntu-latest
-    name: (${{ matrix.target }})
+    name: Rust Bindings - ${{ matrix.settings.name }}
+    runs-on: ${{ matrix.settings.runner }}
     strategy:
+      fail-fast: false
       matrix:
-        target: [
-          x86_64-unknown-linux-gnu,
-          aarch64-unknown-linux-gnu,
-        ]
+        settings:
+          - name: MacOS arm64
+            target: aarch64-apple-darwin
+            runner: macos-latest
+          - name: Linux x86
+            target: x86_64-unknown-linux-gnu
+            runner: ubuntu-latest
+          - name: Linux arm64
+            target: aarch64-unknown-linux-gnu
+            runner: ubuntu-latest
+            use-docker: true
+          - name: Windows x86
+            target: x86_64-pc-windows-msvc
+            runner: windows-latest
+          # - name: Windows arm64
+          #   target: aarch64-pc-windows-msvc
+          #   runner: windows-latest
+          #   use-docker: true
```

Comment:
I had a hard time finding a container that was easy to setup for building Windows on arm64.  I suppose if someone wants in the future should be easy enough to turn on but ran out of time and just commented this case out.